client/new-stack-editor/lib/views/editor.coffee

```diff
@@ -18,15 +21,40 @@ module.exports = class Editor extends BaseView
     @_getSession().setValue content
 
 
+  addContent: (content) -> @ready =>
+
+    session = @_getSession()
+    [ row, column ] = [ session.getLength(), 0 ]
+    session.insert { row, column }, "#{content}\n"
+    @scrollToBottom()
+
+
+  scrollToBottom: -> @ready =>
+
+    { editor } = @aceView.ace
+    line = @_getSession().getLength()
+    editor.scrollToLine line, yes, yes, kd.noop
+    editor.gotoLine line, 0, yes
+
+
   focus: -> @ready =>
     @aceView.ace.focus()
 
 
+  setReadOnly: (state) -> @ready =>
+
+    @aceView.ace.setReadOnly state, no
+    state = if state then 'none' else 'block'
+    @aceView.ace.editor.renderer.$cursorLayer.element.style.display = state
```

Comment:
this is the only way to hide cursor on Ace, it doesn't look good I agree. there are terrible solutions for that one believe me :) https://github.com/ajaxorg/ace/issues/266